### PR TITLE
Taslima week 7

### PR DESCRIPTION
**Pull request by Taslima26**
# Your Details

Your Name:Taslima
Your City:Manchester
Your Slack Name:Tas

# Homework Details

Module:Java script module 3 week -1
Week:

### CODE REVIEW

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 1]
```diff
@@ -18,20 +18,52 @@ The second skill, problem solving, is the most important one. If you get good at
...
-[Lessons in problem solving](https://www.freecodecamp.org/news/how-to-think-like-a-programmer-lessons-in-problem-solving-d1d8bf1de7d2/)
-[Computational thinking](https://www.youtube.com/watch?v=qbnTZCj0ugI)
+- [How to THINK like a programmer](https://www.youtube.com/watch?v=NNazO2tMHno)
+- [Lessons in problem solving](https://www.freecodecamp.org/news/how-to-think-like-a-programmer-lessons-in-problem-solving-d1d8bf1de7d2/)
+- [Computational thinking](https://www.youtube.com/watch?v=qbnTZCj0ugI)
 
 After you've watched these videos I'd like you to answer these questions
 
 ## 1. What do you think the most important quality for a programmer is?
+problem solving..
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:28]: This is definitely one of the top things!

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 2]
```diff
@@ -18,20 +18,52 @@ The second skill, problem solving, is the most important one. If you get good at
...
+
 
 <!-- Write your answer here -->
 
 ## 2. When trying to solve a challenge, what should you do first?
+read and re-read problem untill you completly understand.
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:34]: Some great points here - it is very to easy to mis-read something so it is great practice to go back and double check your problem. And I always find that drawing a diagram helps me to figure things out.

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 3]
```diff
@@ -18,20 +18,52 @@ The second skill, problem solving, is the most important one. If you get good at
...
+If problem seems unsovable ,decide what you can do first with resources and knowledge you already
+have.Devide problem in to  a small managable chunks.
+Start with initial first little step you have.
+Have some sort of mind map,diagram whatever you are comfertable with.
 
 <!-- Write your answer here -->
 
 ## 3. What should you do if you get stuck?
+Take a break,go for walk have some wateror may be some choclates !! and fresh air.
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:43]: Chocolates definitely 😄

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 4]
```diff
@@ -18,20 +18,52 @@ The second skill, problem solving, is the most important one. If you get good at
...
+Rethink problem from begining.
+Some people use rubber duck --means talking out problem loud,Not my thing though.
+Write down problem in your own word and re-read it.
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:46]: I definitely try to rephrase the problem 👍

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 5]
```diff
@@ -18,20 +18,52 @@ The second skill, problem solving, is the most important one. If you get good at
...
 
 <!-- Write your answer here -->
+
+
+problem solving--
+see the problem--
+structure-you need framework
+common framework-
+1.understand the problem---
+draw mind map -visualize the problem-draw diagram,rubber ducky -talk about loud...
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:57]: Some great suggestions here - nice one, Tas 👍

File: week-7/Homework/mandatory/1-debugging-practice/script.js [hunk 6]
```diff
@@ -32,13 +32,16 @@ function submit() {
...
+    author.value == null ||
+    author.value==""
+      
   ) {
     alert("Please fill all fields!");
     return false;
   } else {
-    let book = new Book(title.value, title.value, pages.value, check.checked);
-    library.push(book);
+    let book = new Book(title.value, author.value, pages.value, check.checked);
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/1-debugging-practice/script.js:43]: 👍

File: week-7/Homework/mandatory/1-debugging-practice/script.js [hunk 7]
```diff
@@ -54,7 +57,7 @@ function render() {
   let table = document.getElementById("display");
   let rowsNumber = table.rows.length;
   //delete old table
-  for (let n = rowsNumber - 1; n > 0; n-- {
+  for (let n = rowsNumber - 1; n > 0; n--) {
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/1-debugging-practice/script.js:60]: 👍

File: week-7/Homework/mandatory/1-debugging-practice/script.js [hunk 8]
```diff
@@ -77,9 +80,9 @@ function render() {
     cell4.appendChild(changeBut);
     let readStatus = "";
     if (myLibrary[i].check == false) {
-      readStatus = "Yes";
-    } else {
       readStatus = "No";
+    } else {
+      readStatus = "Yes";
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/1-debugging-practice/script.js:85]: 👍

File: week-7/Homework/mandatory/1-debugging-practice/script.js [hunk 9]
```diff
@@ -32,13 +32,16 @@ function submit() {
     title.value == null ||
     title.value == "" ||
     pages.value == null ||
-    pages.value == ""
+    pages.value == "" ||
+    author.value == null ||
```
Comment (Dedekind561) [inline week-7/Homework/mandatory/1-debugging-practice/script.js:36]: Good well spotted for this one !

**Comment by Taslima26** [comment]
Thank you so much Mitch for larking my work .

**Pull request closed without merging**